### PR TITLE
568175: Correct deserialization

### PR DESCRIPTION
**Pull request by andyreidz**
https://internal.almoctane.com/ui/?p=131002/6001#/entity-navigation?entityType=work_item&id=568175

### CODE REVIEW

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializer.java [hunk 1]
```diff
@@ -78,57 +77,57 @@ public DocumentWorkerDocument deserialize(
...
-                final String field = jsonParser.getCurrentName();
-                jsonParser.nextToken();
-                switch (field) {
-                    case "reference": {
-                        documentWorkerDocument.reference = jsonParser.getValueAsString();
-                        break;
-                    }
-                    case "fields": {
-                        documentWorkerDocument.fields = deserializationContext.readValue(
+            if (jsonParser.currentToken() != JsonToken.FIELD_NAME) {
```
Comment (kusumaghoshdastidar) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializer.java:93]: Is this check necessary? A Json parsing exception might be thrown earlier on @ line 90 :jsonParser.nextToken();

**Review by kusumaghoshdastidar — CHANGES_REQUESTED** [review]
Check if "Expected Field" check is required...add a test if needed.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializer.java (hunk 1 above)
Comment (andyreidz) [reply worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializer.java:93]: This is protecting against bugs, prior versions of this change were incorrectly skipping elements and this check is protecting against that.

**Comment by andyreidz** [comment]
> Check if "Expected Field" check is required...add a test if needed.


> Check if "Expected Field" check is required...add a test if needed.

Added a comment (to the code) to explain

**Pull request merged**